pkg/broker/docker_manager.go

```diff
@@ -0,0 +1,88 @@
+package broker
+
+import (
+	"context"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/cloudfoundry-incubator/blockhead/pkg/config"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/pborman/uuid"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+type dockerContainerManager struct {
+	client DockerClient
+	logger lager.Logger
+}
+
+func NewDockerContainerManager(logger lager.Logger, client DockerClient) ContainerManager {
+	return dockerContainerManager{
+		client: client,
+		logger: logger.Session("docker-container-manager"),
+	}
+}
+
+func (dc dockerContainerManager) Provision(service *config.Service, plan *config.Plan) (brokerapi.ProvisionedServiceSpec, error) {
+	ctx := context.Background()
+	_, err := dc.client.ImagePull(ctx, plan.Image, types.ImagePullOptions{})
+	if err != nil {
+		return brokerapi.ProvisionedServiceSpec{}, err
+	}
```

Comment:
do we actually need to pull, or are we including it for direct comparison to ruby-version?

Reply:
I dont think the ruby version pulls every time. In fact i think it only pulls when we register the service using the `bosh errand`. We can get rid of `ImagePull` here but we need to have a way (possibly better than what the ruby version does) to update the image if/when needed.

Reply:
okay, needs a comment to this effect.